tests/components/threshold/test_binary_sensor.py

```diff
@@ -585,3 +585,215 @@ async def test_device_id(
     utility_meter_entity = entity_registry.async_get("binary_sensor.threshold")
     assert utility_meter_entity is not None
     assert utility_meter_entity.device_id == source_entity.device_id
+
+
+async def test_set_lower_threshold_service(hass: HomeAssistant) -> None:
+    """Test the set_lower_threshold service."""
+    config_entry = MockConfigEntry(
+        data={},
+        domain=DOMAIN,
+        options={
+            CONF_ENTITY_ID: "sensor.test_monitored",
+            CONF_HYSTERESIS: 0.0,
+            CONF_LOWER: 10.0,
+            CONF_NAME: "Test Threshold",
+            CONF_UPPER: None,
+        },
+        title="Test Threshold",
+    )
+    config_entry.add_to_hass(hass)
+
+    # Setup the sensor being monitored
+    hass.states.async_set("sensor.test_monitored", 15)
+
+    assert await hass.config_entries.async_setup(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    # Initial state should be OFF (15 is above lower threshold of 10)
+    state = hass.states.get("binary_sensor.test_threshold")
+    assert state.state == STATE_OFF
+    assert state.attributes[ATTR_LOWER] == 10.0
+
+    # Call the service to set lower threshold to 20
+    await hass.services.async_call(
+        DOMAIN,
+        "set_lower_threshold",
+        {"entity_id": "binary_sensor.test_threshold", "value": 20.0},
```

Comment:
Use const: ATTR_ENTITY_ID, ATTR_VALUE
set_lower_threshold should be added as a StrEnum or const to binary sensor